src/web/services/course.service.ts

```diff
@@ -179,13 +179,19 @@ export class CourseService {
   /**
    * Join a course by calling API.
    */
-  joinCourse(regKey: string, entityType: string, institute: string, institutemac: string): Observable<any> {
+  joinCourse(regKey: string, entityType: string, institute: string,
+      institutemac: string, timezone?: string): Observable<any> {
     const paramMap: Record<string, string> = {
       key: regKey,
       entitytype: entityType,
       instructorinstitution: institute,
       mac: institutemac,
     };
+
+    if (timezone) {
```

Comment:
what if timezone is invalid? I think we should have a validation function for timezone on the front-end side

Reply:
Thanks for the review! I don't think timezone can be invalid on the front-end side since the timezone is taken straight from the timezone service and not from the url, so users have no way of modifying it unless I am missing something here. 

```js
// Relevant section of code
 if (queryParams.isnewinstructoraccount === 'true') {
    this.timezone = this.timezoneService.guessTimezone();
 }
```

`If (timezone)`  on line 191 should handle cases where timezone is an empty string (because timezone cannot be guessed correctly).

Agreed that i shouldn't have assumed the input is valid in the backend though. Would adding validation on just the backend be sufficient in this case?

Reply:
Yes backend validation should be sufficient as we process the timezone in backend only